bodo/pandas/_plan.cpp

```diff
@@ -248,12 +255,58 @@ duckdb::unique_ptr<duckdb::LogicalProjection> make_projection(
         duckdb::make_uniq<duckdb::LogicalProjection>(
             table_idx, std::move(projection_expressions));
 
-    // Add the sources to be joined.
+    // Add the source of the projection.
     proj->children.push_back(std::move(source_duck));
 
     return proj;
 }
 
+duckdb::unique_ptr<duckdb::LogicalAggregate> make_aggregate(
+    std::unique_ptr<duckdb::LogicalOperator> &source, duckdb::idx_t group_index,
+    duckdb::idx_t aggregate_index,
+    std::vector<std::unique_ptr<duckdb::Expression>> &expr_vec,
+    PyObject *out_schema_py) {
+    // Convert std::unique_ptr to duckdb::unique_ptr.
+    auto source_duck = to_duckdb(source);
+    std::vector<duckdb::ColumnBinding> source_cols =
+        source_duck->GetColumnBindings();
+
+    std::vector<duckdb::unique_ptr<duckdb::Expression>> aggregate_expressions;
+    for (auto &expr : expr_vec) {
+        // Convert std::unique_ptr to duckdb::unique_ptr.
+        auto expr_duck = to_duckdb(expr);
+        duckdb::ExpressionClass expr_class = expr_duck->GetExpressionClass();
+
+        switch (expr_class) {
+            case duckdb::ExpressionClass::BOUND_COLUMN_REF: {
+                // Convert the base duckdb::Expression node to its actual
+                // derived type.
+                duckdb::unique_ptr<duckdb::BoundColumnRefExpression> bce =
+                    dynamic_cast_unique_ptr<duckdb::BoundColumnRefExpression>(
+                        std::move(expr_duck));
+                duckdb::ColumnBinding binding = bce->binding;
+                bce->binding = source_cols[binding.column_index];
```

Comment:
Comment on why updating the binding field is necessary here?

Reply:
done.